Quest-4/Part-2/main.py

```diff
@@ -0,0 +1,43 @@
+import re
+
+def main():
+    fileName = "input.txt"
+    lines = readFile(fileName)
+
+    cards = getcards(lines)
+
+    for card in cards:
+        win = calculateWin(card["winNumbers"], card["cardNumbers"])
+        if win != 0:
+            for i in range(card["cardNumber"], card["cardNumber"] + win):
+                cards.append(cards[i])
+
+    totalWin = len(cards)
+    print(totalWin)
+
+def getcards(lines):
+    cards = []
+
+    for line in lines:
+        card = line.rstrip("\n").split(': ')[1]
+        cardNumber = int(re.findall(r'\d+', line.rstrip("\n").split(': ')[0])[0])
+        winNumbers = re.findall(r'\d+', card.split(" | ")[0])
+        cardNumbers = re.findall(r'\d+', card.split(" | ")[1])
```

Comment:
this is somewhat weird. You could just split again on spaces, no? There's a special case for split with no parameters, check it out:
https://docs.python.org/3.3/library/stdtypes.html#str.split